### DISCUSSION TITLE
Empty PNG - Does minecraft need to be installed for this to work?

### DISCUSSION THREAD

**Discussion by DevanshRathi1** [spoutn1k/mcmap#83] [Q&A] [answered]
Trying to run this on a server on DigitalOcean on an ubuntu machine and it was outputting an empty png (no errors though). So before creating an issue i just wanted to know if we need Minecraft installed for this to work because it worked fine on my windows machine.

**Comment by spoutn1k** [comment]
It does not, all mcmap requires is a save file ! This might be a format error, which version are you running ?

**Comment by DevanshRathi1** [comment]
I have tried both cloning the repo and compiling the solution myself and v3.0.1 pre-compiled from the releases. They both have the same empty png output. And the savefile is from minecraft version 1.18.1. Let me know if there is an info i can provide to get this working thanks!

**Comment by spoutn1k** [comment, marked as answer]
Hey so what you need to do first is check the map is not empty. I know this sounds dumb, but if it is a server, maybe the files have not been written; just make sure the region folder is not empty and the files contain data.

Then, run `mcmap` with the `-vv` flag. This enables the heavy debug and might provide explanations for this behaviour. Please paste it here so that I can look at it and we will open an error accordingly.

**Comment by DevanshRathi1** [comment]
Hey so that was the issue. We just needed to roam around the map to load all the .mca files in the region folder. So i had a follow up question. I see that you just need the region files and the level.dat file. I just was curious why you use the level.dat file? Is it just to check the minecraft version or is there some more need?

**Reply by spoutn1k** [reply]
Perfect !

I use the `level.dat` to get the real name of the world. It differs from the folder name and makes it clearer in the debug output. The big idea was to use more of the data in there to:
- Tag the files with the savename: all the files created by mcmap have annotations that describe the software version and coordinates depicted, having the savename in there would have been nice;
- Access the player coordinates and place a marker/center the map around him.

I did not get to writing those in the code yet. I think the `level.dat` is optional either way, so that it does work without it; if that is not the case, then I goofed up.
